### PR TITLE
add tests for Block not downloaded error

### PR DESCRIPTION
**Pull request by sethvincent**
Could probably have a simpler test case but this serves as an example.

### CODE REVIEW

**Comment by gmaclennan** [comment]
I'm not at my computer but does this replicate the same error we are seeing on bugsnag? I think what we are seeing is caused by the equivalent of sparse replication leaving a core with undownloaded blocks (because replication is non-linear so when it is interrupted it leaves the core sparse). I think to replicate that situation we need to replicate a core in sparse and download, say, only blocks 0,1,2,4,5, leaving 3 undownloaded.

**Comment by sethvincent** [comment]
I revised things and it looks like the desired behavior of continued indexing does happen after this error as long as the error emitter is handled.

**Comment by gmaclennan** [comment]
Great, so this should maybe be a passing test? "Indexing continues ignoring missing block" or something similar? It seems like there is no fix that needs made here.

**Comment by sethvincent** [comment]
:+1: yeah, that seems right to me. i'll push a commit to make it pass.

File: .github/workflows/node.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+name: Node.js CI
+
+on:
+  push:
+    branches: [main]
```
Comment (gmaclennan) [inline .github/workflows/node.yml]:
```suggestion
    branches: [master]
```

File: .github/workflows/node.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  pull_request:
+    branches: [main]
```
Comment (gmaclennan) [inline .github/workflows/node.yml]:
```suggestion
    branches: [master]
```

**Review by gmaclennan — CHANGES_REQUESTED** [review]

**Review by gmaclennan — APPROVED** [review]

**Pull request merged**